fastly/block_fastly_service_v1_logging_cloudfiles_test.go

```diff
@@ -0,0 +1,317 @@
+		PublicKey:         pgpPublicKey(t),
+		Format:            "%h %l %u %t \"%r\" %>s %b %T",
+		FormatVersion:     2,
+		GzipLevel:         1,
+		MessageType:       "blank",
+		Path:              "new/",
+		Region:            "LON",
+		Period:            3601,
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+		TimestampFormat:   "%Y-%m-%dT%H:%M:%S.000",
+	}
+
+	log2 := gofastly.Cloudfiles{
+		Version:           1,
+		Name:              "another-cloudfiles-endpoint",
+		BucketName:        "bucket2",
+		User:              "user2",
+		AccessKey:         "secret2",
+		PublicKey:         pgpPublicKey(t),
+		Format:            "%h %l %u %t \"%r\" %>s %b",
+		FormatVersion:     2,
+		GzipLevel:         0,
+		MessageType:       "classic",
+		Path:              "two/",
+		Region:            "SYD",
+		Period:            3600,
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+		TimestampFormat:   "%Y-%m-%dT%H:%M:%S.000",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1CloudfilesConfig(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.none", &service),
+					testAccCheckFastlyServiceV1CloudfilesAttributes(&service, []*gofastly.Cloudfiles{&log1}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.none", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.none", "logging_cloudfiles.#", "1"),
+				),
+			},
+
+			{
+				Config: testAccServiceV1CloudfilesConfig_update(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.none", &service),
+					testAccCheckFastlyServiceV1CloudfilesAttributes(&service, []*gofastly.Cloudfiles{&log1_after_update, &log2}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.none", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.none", "logging_cloudfiles.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFastlyServiceV1CloudfilesAttributes(service *gofastly.ServiceDetail, cloudfiles []*gofastly.Cloudfiles) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		conn := testAccProvider.Meta().(*FastlyClient).conn
+		cloudfilesList, err := conn.ListCloudfiles(&gofastly.ListCloudfilesInput{
+			Service: service.ID,
+			Version: service.ActiveVersion.Number,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error looking up Cloudfiles Logging for (%s), version (%d): %s", service.Name, service.ActiveVersion.Number, err)
+		}
+
+		if len(cloudfilesList) != len(cloudfiles) {
+			return fmt.Errorf("Cloudfiles List count mismatch, expected (%d), got (%d)", len(cloudfiles), len(cloudfilesList))
+		}
+
+		log.Printf("[DEBUG] cloudfilesList = %#v\n", cloudfilesList)
```

Comment:
Is this still needed?

Reply:
This could still help with debugging global state.